packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -642,39 +672,36 @@ class _HeroFlight {
       // path to be the same (in reverse) as the push flight path.
       _proxyAnimation.parent = ReverseAnimation(newManifest.animation);
       heroRectTween = ReverseTween<Rect?>(heroRectTween);
-    } else if (manifest!.type == HeroFlightDirection.pop && newManifest.type == HeroFlightDirection.push) {
+    } else if (manifest.type == HeroFlightDirection.pop && newManifest.type == HeroFlightDirection.push) {
       // A pop flight was interrupted by a push.
       assert(newManifest.animation.status == AnimationStatus.forward);
-      assert(manifest!.toHero == newManifest.fromHero);
-      assert(manifest!.toRoute == newManifest.fromRoute);
+      assert(manifest.toHero == newManifest.fromHero);
+      assert(manifest.toRoute == newManifest.fromRoute);
 
       _proxyAnimation.parent = newManifest.animation.drive(
         Tween<double>(
-          begin: manifest!.animation.value,
+          begin: manifest.animation.value,
           end: 1.0,
         ),
       );
-      if (manifest!.fromHero != newManifest.toHero) {
-        manifest!.fromHero.endFlight(keepPlaceholder: true);
+      if (manifest.fromHero != newManifest.toHero) {
+        manifest.fromHero.endFlight(keepPlaceholder: true);
         newManifest.toHero.startFlight();
-        heroRectTween = _doCreateRectTween(
-            heroRectTween.end,
-            _boundingBoxFor(newManifest.toHero.context, newManifest.toRoute.subtreeContext),
-        );
+        heroRectTween = manifest.createHeroRectTween(begin: heroRectTween.end, end: newManifest.toHeroLocation);
       } else {
         // TODO(hansmuller): Use ReverseTween here per github.com/flutter/flutter/pull/12203.
-        heroRectTween = _doCreateRectTween(heroRectTween.end, heroRectTween.begin);
+        heroRectTween = manifest.createHeroRectTween(begin: heroRectTween.end, end: heroRectTween.begin);
```

Comment:
Does this matter? This will only affect when the divert happens when toHero is right at the hero rect tween end. If that is the case, changing the new manifest may result in a sudden jump.